app/amity.py

```diff
@@ -0,0 +1,466 @@
+        role = role.upper()
+        accomodation = accomodation
+
+        if full_name in self.all_people:
+            return ("%s already used " % full_name)
+        else:
+            if(role in ("FELLOW", "STAFF")):
+                self.all_people.append(full_name)
+
+                # assign random office to everyone
+                office = self.select_random_office()
+                if office is False:
+                    self.office_waiting_list.append(full_name)
+                    msg = "No office available at this time"
+                    print(msg)
+                else:
+                    self.room_allocations["office"][office].append(full_name)
+                    msg = "Allocated office " + office
+                    print(msg)
+
+                if role == "FELLOW":
+                    self.all_fellow.append(full_name)
+                    msg = ("%s - fellow added successfully" % full_name)
+                    print(msg)
+
+                    # if fellow accomodation is yes allocate living
+                    if accomodation == "Y":
+                        living = self.select_random_living()
+                        if living is False:
+                            self.living_space_waiting_list.append(full_name)
+                            msg2 = "No Living space available at this time"
+                        else:
+                            self.room_allocations["living"][living].append(
+                                full_name)
+                            msg2 = "Allocated living space " + living
+                        print(msg2)
+
+                elif role == "STAFF":
+                    self.all_staff.append(full_name)
+                    msg = ("%s - staff added successfully" % full_name)
+                    print(msg)
+                    if accomodation == "Y":
+                        msg2 = "Sorry! staff don't get accomodation"
+                        print(msg2)
+
+            else:
+                print("Error! Please indicate correct role")
+
+    def select_random_office(self):
+        available = []
+        for office in self.all_offices:
+            if len(self.room_allocations["office"][office]) < 6:
+                available.append(office)
+        if len(available) < 1:
+            return False
+        office = choice(available)
+        return(office)
+
+    def select_random_living(self):
+        available = []
+        for living in self.all_living:
+            if len(self.room_allocations["living"][living]) < 4:
+                available.append(living)
+        if len(available) < 1:
+            return False
+        living = choice(available)
+        return(living)
+
+    def if_was_assigned(self, full_name, room_type):
+        rooms = []
+        for x in self.room_allocations[room_type].keys():
+            rooms.append(x)
+        for room in rooms:
+            if full_name in self.room_allocations[room_type][room]:
+                return(room)
+
```

Comment:
the rooms variable is doing something quite redudant.

```
 for room_x in self.room_allocations[room_type].keys():
      if full_name in self.room_allocations[room_type][room_x]:
                return(room_x)
```
looks you could do away with one of the for loops.